src/cache-fs.ts

```diff
@@ -97,9 +147,54 @@ export class CacheFs implements FileSystem {
         });
     }
 
+    private rescanOnError() {
+        this.cache.loadDirectoryTree().then(cachedTree => {
+            this.isTreeCached = false;
+            this.loadDirectoryTree().then(realTree => {
+                const {toDelete, toAdd, toChange} = getTreesDiff(
+                    nodesToMap(cachedTree.children),
+                    nodesToMap(realTree.children)
+                );
+
```

Comment:
Need to take into account what happens with changes that are both in the tree diff, *and* are reported from the underlying events.
It's like Idempotency, any second identical consecutive event on the same file should be dropped. so if we get a `fileChanged` or `fileCreated` event with content matching what we already have in cache (the latest version of the file which was reported to the user) we need to silently ignore that event. similar principle for the rest of the events.